### ISSUE TITLE
Warning not reported for package private classes inside source file of other class

### ISSUE THREAD

**Issue by arend-von-reinersdorff** [SonarQubeCommunity/sonar-findbugs#53]
I think I found a problem where the SonarQube Findbugs plugin doesn't report warnings in package private classes. Only occurs if the class is inside the source file of another class and if it's not an inner class. So it's probably a rare problem.

**Reproduce**
Analyze a source file with a public class and a package private class. For example CompilationUnitHasPackageClass.java:
```
public class CompilationUnitHasPackageClass implements Comparable<CompilationUnitHasPackageClass> {
    public int compareTo(CompilationUnitHasPackageClass o) {
        throw new UnsupportedOperationException();
    }
}

class PackageClass implements Comparable<PackageClass> {
    public int compareTo(PackageClass o) {
        throw new UnsupportedOperationException();
    }
}
```

I analyzed with SonarQube Findbugs plugin version 3.4.3 and SonarQube 5.6.1.

**Result**
- (Warning EQ_COMPARETO_USE_OBJECT_EQUALS is correctly reported for CompilationUnitHasPackageClass.compareTo())
- No warning reported for PackageClass.compareTo()
- Warning logged:

> [WARNING] The class 'PackageClass' could not be match to its original source file. It might be a dynamically generated class.

**Expected**
- Warning EQ_COMPARETO_USE_OBJECT_EQUALS is reported for PackageClass.compareTo()
- No Warning logged

**Comment by h3xstream** [comment]
This issue should be fix (#40)

You can download the release candidate and ask to reopen the issue if anything is found.
https://github.com/SonarQubeCommunity/sonar-findbugs/releases/tag/3.4.4

**Comment by arend-von-reinersdorff** [comment]
Sorry, I missed the existing issue #40. I tried out the 3.4.4 release candidate and it fixes the issue. Thanks :-)
